### PR TITLE
Prevented some _all_docs cases truncating the response

### LINKED ISSUE #411: Wrong results from AllDocsRequest interface getResponse().getIdsAndRevs()
Please include the following information in your ticket.

- CloudantClient (java-cloudant) version(s) that are affected by this issue.
I tried 2.9 and 2.11
- Java version (including vendor and platform).
java version "1.8.0_144"
- If you're using the optional okhttp dependency.
'com.squareup.okhttp3:okhttp:3.9.0'
- A small code sample that demonstrates the issue.

```
Database db = client.database(database, false);
    AllDocsRequest allDocsRequest = db.getAllDocsRequestBuilder()
                                      .keys(ids)
                                      .build();

    try {
      return allDocsRequest.getResponse().getIdsAndRevs();
    }  catch (Exception e) {

    }
```

create 4 docs in a database, with ids e.g. k_1, k_2, k_3, k_4, now remove k_3 and k_4,
request _all_docs interface from a browser or postman chrome app, I got a full list of 4 items, with k_3 and k_4 marked as deleted
with Cloudant-client java lib, I got 3 items back

cloudant java lib defect - 1
cloudant java lib defect - 2

### PR DESCRIPTION
**Pull request by ricellis**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/java-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.md](https://github.com/cloudant/java-cloudant/blob/master/CHANGES.md) 
- [x] You have completed the PR template below:

## What

Prevented some `_all_docs` cases truncating the last response row.

## How

Fixed #411 where `total_rows` < `rows` length in an `_all_docs` response fell into the pagination branch causing a truncated response.
Stopped using `total_rows` as `rowsPerPage` in `ViewResponseImpl`.
Updated `CHANGES.md`.

## Testing

Added a test for 2 deleted, 2 undeleted docs returned to a `keys`
`_all_docs` request.

## Issues

Fixes #411

### CODE REVIEW

File: cloudant-client/src/main/java/com/cloudant/client/internal/views/ViewResponseImpl.java [hunk 1]
```diff
@@ -80,12 +80,12 @@
...
 
-        long rowsPerPage = (initialQueryParameters.getRowsPerPage() != null) ?
-                initialQueryParameters.getRowsPerPage().longValue() : totalRows;
+        Long rowsPerPage = (initialQueryParameters.getRowsPerPage() != null) ?
+                initialQueryParameters.getRowsPerPage().longValue() : null;
 
-        //we expect limit = rowsPerPage + 1 results, if we have rowsPerPage or less we are on the
-        // last page
-        hasNext = resultRows > rowsPerPage;
+        // We expect limit = rowsPerPage + 1 results, if we have rowsPerPage or less or rowsPerPage
```
Comment (tomblench) [inline cloudant-client/src/main/java/com/cloudant/client/internal/views/ViewResponseImpl.java]: Comment is now a bit confusing, how about

"// We expect limit = rowsPerPage + 1 results, if we have rowsPerPage or less, or if rowsPerPage wasn't set then we are on the last page"

File: cloudant-client/src/test/java/com/cloudant/tests/ViewsTest.java [hunk 2]
```diff
@@ -1246,9 +1247,6 @@ public void staleParameterValues() throws Exception {
         UnpaginatedRequestBuilder<String, Integer> viewBuilder = database.getViewRequestBuilder
                 ("testDDoc", "testView").newRequest(Key.Type.STRING, Integer.class);
 
-        MockResponse mockResponse = new MockResponse().setResponseCode(200).setBody
```
Comment (tomblench) [inline cloudant-client/src/test/java/com/cloudant/tests/ViewsTest.java]: Why have these mock responses been removed?

Comment (ricellis) [reply cloudant-client/src/test/java/com/cloudant/tests/ViewsTest.java]: The same content was being enqueued in the `assertStaleParameter` method. So they ended being queued twice. This left extra requests in the mockwebserver that caused the new mock test to fail.

Comment (tomblench) [reply cloudant-client/src/test/java/com/cloudant/tests/ViewsTest.java]: OK, understood

File: cloudant-client/src/main/java/com/cloudant/client/internal/views/ViewResponseImpl.java (hunk 1 above)
Comment (ricellis) [reply cloudant-client/src/main/java/com/cloudant/client/internal/views/ViewResponseImpl.java]: Fixed in c2da7f8

**Pull request merged**